Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/CqlCompiler.java

```diff
@@ -0,0 +1,255 @@
+    private List<CqlTranslatorException> warnings = null;
+    private List<CqlTranslatorException> messages = null;
+    private VersionedIdentifier sourceInfo = null;
+    private NamespaceInfo namespaceInfo = null;
+    private ModelManager modelManager = null;
+    private LibraryManager libraryManager = null;
+    private UcumService ucumService = null;
+
+    public CqlCompiler(ModelManager modelManager, LibraryManager libraryManager) {
+        this(null, null, modelManager, libraryManager, null);
+    }
+
+    public CqlCompiler(NamespaceInfo namespaceInfo, ModelManager modelManager, LibraryManager libraryManager) {
+        this(namespaceInfo, null, modelManager, libraryManager, null);
+    }
+
+    public CqlCompiler(NamespaceInfo namespaceInfo, ModelManager modelManager,
+                        LibraryManager libraryManager, UcumService ucumService) {
+        this(namespaceInfo, null, modelManager, libraryManager, ucumService);
+    }
+
+    public CqlCompiler(NamespaceInfo namespaceInfo, VersionedIdentifier sourceInfo, ModelManager modelManager,
+                        LibraryManager libraryManager, UcumService ucumService) {
+        this.sourceInfo = sourceInfo;
+        this.namespaceInfo = namespaceInfo;
+        this.modelManager = modelManager;
+        this.libraryManager = libraryManager;
+        this.ucumService = ucumService;
+
+        if (this.sourceInfo == null) {
+            this.sourceInfo = new VersionedIdentifier().withId("Anonymous").withSystem("text/cql");
+        }
+
+        if (this.namespaceInfo != null) {
+            modelManager.getNamespaceManager().ensureNamespaceRegistered(this.namespaceInfo);
+            libraryManager.getNamespaceManager().ensureNamespaceRegistered(this.namespaceInfo);
+        }
+
+        if (libraryManager.getNamespaceManager().hasNamespaces() && libraryManager.getLibrarySourceLoader() instanceof NamespaceAware) {
+            ((NamespaceAware)libraryManager.getLibrarySourceLoader()).setNamespaceManager(libraryManager.getNamespaceManager());
+        }
+
+        if (libraryManager.getUcumService() == null) {
+            libraryManager.setUcumService(this.ucumService);
+        }
+    }
+
+    public Library getLibrary() {
+        return library;
+    }
+    public TranslatedLibrary getTranslatedLibrary() {
+        return translatedLibrary;
+    }
+    public Object toObject() {
+        return visitResult;
+    }
+    public List<Retrieve> toRetrieves() {
+        return retrieves;
+    }
+    public Map<String, TranslatedLibrary> getTranslatedLibraries() {
+        return libraryManager.getTranslatedLibraries();
+    }
+
+    public Map<String, Library> getLibraries() {
+        Map<String, Library> result = new HashMap<String, Library>();
+        for (String libraryName : libraryManager.getTranslatedLibraries().keySet()) {
+            result.put(libraryName, libraryManager.getTranslatedLibraries().get(libraryName).getLibrary());
```

Comment:
Similar to the above comment, this is now "getCompiledLibraries" or "getElmLibraries".

Reply:
Converted TranslatedLibrary to CompiledLibrary